test/test-generate-kmp.ts

```diff
@@ -13,17 +13,37 @@ const sources = [
     ["SEN", 4816],
   ]),
 ];
-const modelId = "nrc.en.mtnt";
+const modelID = "nrc.en.mtnt";
 
 test("it should generate a kmp file", async (t) => {
-  const kmp = await generateKmp(langName, bcp47Tag, sources, modelId);
+  const kmp = await generateKmp(langName, bcp47Tag, sources, modelID);
   t.assert(kmp.byteLength > 0);
 
   const new_zip = new JSZip();
   const zip = await new_zip.loadAsync(kmp);
   const isKmpExist = zip.file("kmp.json");
-  const isModelExist = zip.file("nrc.en.mtnt.model.js");
+  const isModelExist = zip.file(modelID + ".model.js");
 
   t.assert(isKmpExist != null);
   t.assert(isModelExist != null);
 });
+
+const modelID2 = "national_research_council_canada.str.sencoten";
+
+test("The modelID should determines the filename of the model.js file in kmp archive", async (t) => {
+  const kmp = await generateKmp(langName, bcp47Tag, sources, modelID2);
+  const new_zip = new JSZip();
+  const zip = await new_zip.loadAsync(kmp);
+  const kmpFile = zip.file("kmp.json");
+  const modelFile = zip.file(modelID2 + ".model.js");
+
+  t.assert(kmpFile != null);
+  t.assert(modelFile != null);
+
+  if (kmpFile) {
```

Comment:
I think it's better to fail the test here if `kmpFile` is false-y:

```suggestion
  if (!kmpFile) {
    t.fail("could not load kmp.json from KMP archive");
  }
```

**NOTE**: You'd have to adjust the rest of the test case syntax if you accept this suggestion!